index.bs

```diff
@@ -305,32 +305,46 @@ For the main <dfn>sanitize</dfn> operation, using a {{ParentNode}} |node|, a
          1. If "data-" is a [=code unit prefix=] of [=Attr/local name=] and
             if [=Attr/namespace=] is `null` and
             if |config|["{{SanitizerConfig/dataAttributes}}"] exists and is false:
-            1. Remove |attr| from |child|.
+            1. Then [=remove an attribute|remove=] |attr||.
       1. else if |config|["{{SanitizerConfig/removeAttributes}}"] exists and
          |config|["{{SanitizerConfig/removeAttributes}}"] [=SanitizerConfig/contains=]
          |attrName|:
-         1. Remove |attr| from |child|.
+         1. Then [=remove an attribute|remove=] |attr|.
       1. If |config|["{{SanitizerConfig/elements}}"][|elementName|] exists,
          and if
          |config|["{{SanitizerConfig/elements}}"][|elementName|]["{{SanitizerElementNamespaceWithAttributes/attributes}}"]
          exists, and if
          |config|["{{SanitizerConfig/elements}}"][|elementName|]["{{SanitizerElementNamespaceWithAttributes/attributes}}"]
          does not [=SanitizerConfig/contain=] |attrName|:
-         1. Remove |attr| from |child|.
+         1. Then [=remove an attribute|remove=] |attr|.
       1. If |config|["{{SanitizerConfig/elements}}"][|elementName|] exists,
          and if
          |config|["{{SanitizerConfig/elements}}"][|elementName|]["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"]
          exists, and if
          |config|["{{SanitizerConfig/elements}}"][|elementName|]["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"]
          [=SanitizerConfig/contains=] |attrName|:
-         1. Remove |attr| from |child|.
-      1. If &laquo;[|elementName|, |attrName|]&raquo; matches an entry in the
-         [=navigating URL attributes list=], and if |attr|'s [=protocol=] is
-         "`javascript:`":
-         1. Then remove |attr| from |child|.
+         1. Then [=remove an attribute|remove=] |attr|.
+      1. If the [=navigating URL attributes list=] [=SanitizerConfig/contains=]
+         &laquo;[|elementName|, |attrName|]&raquo; and "`javascript:`" is a
+         [=code unit prefix=] of the result of [=strip leading and
+         trailing ASCII whitespace=] from |attr|'s [=get an attribute
+         value|value=]:
```

Comment:
This doesn't work. We need to actually parse the value as a URL (using the basic URL parser to avoid doing anything with `blob:` URLs) and then look at the scheme.

See https://url.spec.whatwg.org/#concept-basic-url-parser.

This has quite a few loopholes:

* "`JAVAscript:`"
* "`java\tscript:`" (replace `\t` with a literal tab)
* "`\0javascript:`" (replace `\0` with U+0000)

etc.

Reply:
Done.